tasks/system_probe.py

```diff
@@ -563,10 +563,15 @@ def get_ebpf_build_flags(target=None, kernel_release=None, minimal_kernel_releas
     return flags
 
 
+def ebpf_check_source_file(ctx, src_file):
+    ctx.run("if grep -q ' inline ' {src_file}; then echo -e '\u001b[7mPlease use __always_inline instead of inline in {src_file}\u001b[0m';exit 1;fi".format(src_file=src_file), echo=False)
+
+
 def build_network_ebpf_compile_file(
     ctx, parallel_build, build_dir, p, debug, network_prebuilt_dir, network_flags, extension=".bc"
 ):
     src_file = os.path.join(network_prebuilt_dir, f"{p}.c")
+    ebpf_check_source_file(ctx, src_file)
```

Comment:
This only checks the `.c` file and not any of the `.h` files, correct?

Reply:
yes this is correct
the runtime compilation include everything

Reply:
Runtime compilation uses different sources from prebuilt though. Should we just check all `.c` and `.h` files in specific directories? Perhaps at the start of the `build_object_files` function?

Reply:
for runtime it's done in pkg/ebpf/bytecode/runtime/integrity.go
One would be better, but we don't have really one place of truth for source files

Reply:
I'd like to check more than just the top-level `.c` prebuilt files. Can we just scan all `.c` and `.h` files in ebpf source directories? That would cover both prebuilt and runtime.

Reply:
done